File: src/commands/InitCommand.ts
import { init } from "../core/repo/init.js";
import logger from "../common/helpers/logger.js";
import { Command } from "./types.js";

export class InitCommand implements Command {
  execute(_args: string[]): void {
    init();
    logger.log(`Initialized empty minigit repository`);
  }
}

File: src/cli/commands/init_command.cpp
#include <print>

#include "minigit/cli/commands/init_command.hpp"
#include "minigit/core/repo/repository.hpp"

namespace minigit::cli {

    int InitCommand::execute([[maybe_unused]] repo::Repository* repo,
                             [[maybe_unused]] std::span<const std::string_view> args) const
    {
        (void)repo::Repository::init(std::filesystem::current_path());
        std::println("Initialized empty minigit repository");
        return 0;
    }

} // namespace minigit::cli
